File: Medium/2026-04-11-3741-Minimum-Distance-Between-Three-Equal-Elements-II/solution.java
import java.util.*;

public class Solution {
    public int minimumDistance(int[] nums) {
        Map<Integer, List<Integer>> indexMap = new HashMap<>();
        
        for (int i = 0; i < nums.length; i++) {
            indexMap.computeIfAbsent(nums[i], k -> new ArrayList<>()).add(i);
        }
        
        int minDistance = Integer.MAX_VALUE;
        boolean found = false;

        for (List<Integer> indices : indexMap.values()) {
            if (indices.size() >= 3) {
                for (int i = 0; i < indices.size(); i++) {
                    for (int j = i + 1; j < indices.size(); j++) {
                        for (int k = j + 1; k < indices.size(); k++) {
                            int distance = Math.abs(indices.get(i) - indices.get(j)) +
                                           Math.abs(indices.get(j) - indices.get(k)) +
                                           Math.abs(indices.get(k) - indices.get(i));
                            minDistance = Math.min(minDistance, distance);
                            found = true;
                        }
                    }
                }
            }
        }
        
        return found ? minDistance : -1;
    }
}

File: Medium/2026-04-11-3741-Minimum-Distance-Between-Three-Equal-Elements-II/solution.py
class Solution:
    def minimumDistance(self, nums: List[int]) -> int:
        from collections import defaultdict
        indices = defaultdict(list)
        
        for i, num in enumerate(nums):
            indices[num].append(i)
        
        min_distance = float('inf')
        
        for idx_list in indices.values():
            if len(idx_list) < 3:
                continue
            for i in range(len(idx_list) - 2):
                for j in range(i + 1, len(idx_list) - 1):
                    for k in range(j + 1, len(idx_list)):
                        dist = abs(idx_list[i] - idx_list[j]) + abs(idx_list[j] - idx_list[k]) + abs(idx_list[k] - idx_list[i])
                        min_distance = min(min_distance, dist)
        
        return min_distance if min_distance != float('inf') else -1